new-dti-website/components/icons.tsx

```diff
@@ -9,7 +9,7 @@ interface IconProps {
   activeIcon: string;
   altText: string;
   isActive: boolean;
-  onClick: () => void;
+  onClick?: () => void;
```

Comment:
In general, I think we should be using `<button>` for all interactive content.

However, I believe sometimes there might be exceptions. 

Off the top of my head, for example, imagine we have an interactive card with a link inside. From a user's standpoint, they would expect that clicking anywhere on the card (except on the link) triggers an action, and then clicking the link triggers another action. 

We might want to avoid using `<button>` to wrap the card a since there would be a nested `<a>` link inside. Using `<button>` with `<a>` inside of it would be confusing to screenreader technologies and not very accessible.

In this case, I see 2 solutions.

The first would involve quite some CSS. So we'd use something like a regular `<div>` or `<article>`. And then we could have a hidden `<button>` element that we'd style using CSS (with `::after`) to cover the entire card. Then with some `z-index` trickery, we'd move `<a>` on top. The reason we do all of this is to try and keep the `onClick` on the `<button>` element because it is semantic, natively focusable to keyboard keys, and helps screenreader technologies. But this is quite complicated.

Another solution I'd propose would be to just use that regular `<div>` tag, and then apply the `role="button"` to that `<div>`. Then you could combine it with `tabindex="0"` and the `onClick` attribute and basically have yourself a `<button>`. However, we should be wary of this approach because `role="button"` tells the screenreader that the element is a button, but it doesn't provide any actual functionality. Again, this goes back to my point that using a `<button>` is almost always the best case.

So yeah TL;DR, let's try and avoid using `onClick` on an element that isn't a `<button>`. It's simpler, more accessible, and easier to maintain.

Reply:
hope this helps :)